src/docfx/restore/RestoreGit.cs

```diff
@@ -68,53 +68,58 @@ public static async Task GC(string docsetPath, Config config, Func<string, Task>
 
         private static List<(string restoreDir, List<string> hrefs)> GetRestoreItems(string docsetPath, Config config, string locale)
         {
-            // restore dependency repositories
-            var restoreItems = config.Dependencies.Values.GroupBy(d => GetRestoreRootDir(d), PathUtility.PathComparer).Select(g => (g.Key, g.Distinct().ToList())).ToList();
+            var gitDependencies = config.Dependencies.Values.Concat(GetLocRestoreItem(docsetPath, config, locale));
 
-            // restore loc repository
-            var (locRestoreDir, locRepoHref) = GetLocRestoreItem(docsetPath, config, locale);
-            if (!string.IsNullOrEmpty(locRepoHref) && !string.IsNullOrEmpty(locRestoreDir))
-            {
-                restoreItems.Add((locRestoreDir, new List<string> { locRepoHref }));
-            }
-
-            return restoreItems;
+            return gitDependencies.GroupBy(d => GetRestoreRootDir(d), PathUtility.PathComparer).Select(g => (g.Key, g.Distinct().ToList())).ToList();
         }
 
-        private static (string locRestoreDir, string href) GetLocRestoreItem(string docsetPath, Config config, string locale)
+        private static IEnumerable<string> GetLocRestoreItem(string docsetPath, Config config, string locale)
         {
-            // restore loc repository
             if (string.IsNullOrEmpty(locale))
             {
-                return default;
+                yield break;
             }
 
             if (string.Equals(locale, config.Localization.DefaultLocale, StringComparison.OrdinalIgnoreCase))
             {
-                return default;
+                yield break;
             }
 
             if (config.Localization.Mapping != LocalizationMapping.Repository && config.Localization.Mapping != LocalizationMapping.RepositoryAndFolder)
             {
-                return default;
+                yield break;
             }
 
             var repo = Repository.CreateFromFolder(Path.GetFullPath(docsetPath));
             if (repo == null)
             {
-                return default;
+                yield break;
+            }
+
+            if (config.Localization.Bilingual)
+            {
+                // Bilingual repos also depend on non bilingual branch
+                yield return ToHref(LocalizationConvention.GetLocalizationRepo(
```

Comment:
is that too much heavy to calculate non-sxs branch?

Reply:
GetLocalizationRepo should be compute only and runs very fast right? I want a single place to handle `-sxs` logic.